src/main/java/com/chromediopside/service/LoginService.java

```diff
@@ -3,31 +3,33 @@
 import com.chromediopside.datatransfer.ErrorResponse;
 import com.chromediopside.datatransfer.LoginForm;
 import com.chromediopside.datatransfer.TokenResponse;
+import java.util.ArrayList;
+import java.util.List;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
 
 @Component
 public class LoginService {
-  
+
   public ResponseEntity<?> login(LoginForm loginForm) {
-    String missingValues = "";
-    boolean complete = true;
+    if (missingValues(loginForm).equals("")) {
+      return new ResponseEntity<>(new TokenResponse("abc123"), HttpStatus.OK);
+    }
+    return new ResponseEntity<>(new ErrorResponse("Missing parameter(s): "
+        + missingValues(loginForm) + "!"), HttpStatus.BAD_REQUEST);
+  }
+
+
+  private String missingValues(LoginForm loginForm) {
```

Comment:
You should find a more general solution to check if you got all the required parameters. This solution works with just LoginForm objects and with just the already defined parameters.